### PR TITLE
Add collection in querynode when sync new partitions

### PR DESCRIPTION
**Pull request by bigsheeper**
issues: https://github.com/milvus-io/milvus/issues/23824

/kind bug

### CODE REVIEW

File: internal/querycoordv2/job/job_load.go [hunk 1]
```diff
@@ -132,7 +132,7 @@ func (job *LoadCollectionJob) Execute() error {
 	log.Info("find partitions to load", zap.Int64s("partitions", lackPartitionIDs))
 
 	// 2. loadPartitions on QueryNodes
-	err = loadPartitions(job.ctx, job.meta, job.cluster, false, req.GetCollectionID(), lackPartitionIDs...)
+	err = loadPartitions(job.ctx, job.meta, job.cluster, nil, false, req.GetCollectionID(), lackPartitionIDs...)
```
Comment (czs007) [inline internal/querycoordv2/job/job_load.go:135]: It is recommended to pass a real schema

File: internal/querycoordv2/job/job_load.go [hunk 2]
```diff
@@ -309,7 +309,7 @@ func (job *LoadPartitionJob) Execute() error {
 	log.Info("find partitions to load", zap.Int64s("partitions", lackPartitionIDs))
 
 	// 2. loadPartitions on QueryNodes
-	err := loadPartitions(job.ctx, job.meta, job.cluster, false, req.GetCollectionID(), lackPartitionIDs...)
+	err := loadPartitions(job.ctx, job.meta, job.cluster, nil, false, req.GetCollectionID(), lackPartitionIDs...)
```
Comment (czs007) [inline internal/querycoordv2/job/job_load.go:312]: It is recommended to pass a real schema

File: internal/querycoordv2/job/job_sync.go [hunk 3]
```diff
@@ -35,19 +35,22 @@ type SyncNewCreatedPartitionJob struct {
...
 	cluster session.Cluster
+	broker  meta.Broker
 }
 
 func NewSyncNewCreatedPartitionJob(
 	ctx context.Context,
 	req *querypb.SyncNewCreatedPartitionRequest,
 	meta *meta.Meta,
 	cluster session.Cluster,
+	broker meta.Broker,
```
Comment (czs007) [inline internal/querycoordv2/job/job_sync.go:46]: what's this?

File: internal/querycoordv2/job/job_load.go (hunk 1 above)
Comment (bigsheeper) [reply internal/querycoordv2/job/job_load.go:135]: load job doesn't have a schema

**Pull request merged**